src/core/mount.c

```diff
@@ -1750,9 +1783,54 @@ static int mount_load_proc_self_mountinfo(Manager *m, bool set_flags) {
                 if (!device || !path)
                         continue;
 
+                if (ignore_mount(path, fstype))
+                        continue;
+
+                size = snprintf(mount_entry, mount_entry_size, "%s %s %s %s", device, path, fstype, options);
+
+                if ((size_t)size >= mount_entry_size) {
+                        if (mount_entry)
+                                free(mount_entry);
+                        mount_entry_size = 1024 * (size / 1024 + 1);
+                        mount_entry = new(char, mount_entry_size);
+                        if (!mount_entry) {
+                                mount_entry_size = 0;
+                                return -ENOMEM;
+                        }
+                        size = snprintf(mount_entry, mount_entry_size, "%s %s %s %s", device, path, fstype, options);
+                }
```

Comment:
first of all, asprintf() exists. Secondly, please use strjoin() for joining strings, printf() and friends are awfully slow.

But most importantly, this is sloppy: turning this into a string first, and then hashing is unnecessary, you can just hash it directly, just provide a hash_ops implementation, that hashes the four fields as necessary. Grep the tree for `siphash24_compress()` invocations, if you need inspiration for that.

Reply:
ok, indeed this can be enhanced